lib/opensusebasetest.pm

```diff
@@ -556,13 +557,57 @@ sub remount_tmp_if_ro {
     script_run 'touch /tmp/test_ro || mount -t tmpfs /dev/shm /tmp';
 }
 
+=head2 select_serial_terminal
+
+    select_serial_terminal($root);
+
+Select most suitable text console with root user.
+The choice is made by BACKEND and other variables.
+
+Optional C<root> parameter specifies, whether use root user
+(1 or default) or prefer non-root user if available.
+=cut
+sub select_serial_terminal {
+    my ($self, $root) = @_;
+    $root = shift // 1;
+    my $console;
+
+    if (check_var('BACKEND', 'qemu')) {
+        if ($root) {
+            $console = get_var('NO_VIRTIO_CONSOLE') ? 'root-console' : 'root-virtio-terminal';
+        } else {
+            $console = get_var('NO_VIRTIO_CONSOLE') ? 'user-console' : 'virtio-terminal';
+        }
+    } elsif (is_svirt_except_s390x) {
+        $console = 'sut';
+    } elsif (check_var('BACKEND', 'ikvm') || check_var('BACKEND', 'ipmi') || check_var('BACKEND', 'spvm')) {
+        $console = 'root-ssh';
+    } elsif (check_var('VIRSH_VMM_FAMILY', 'hyperv')) {
+        $console = 'hyperv-intermediary';
+    }
+
+    bmwqemu::fctinfo("Selecting console '$console'");
+    select_console($console);
+}
+
+=head2 select_user_serial_terminal
+
+    select_user_serial_terminal();
+
+Select most suitable text console with non-root user.
+The choice is made by BACKEND and other variables.
+=cut
+sub select_user_serial_terminal {
+    select_serial_terminal(0);
+}
+
 # useful post_fail_hook for any module that calls wait_boot and x11_start_program
 ##
 ## we could use the same approach in all cases of boot/reboot/shutdown in case
 ## of wait_boot, e.g. see `git grep -l reboot | xargs grep -L wait_boot`
 sub post_fail_hook {
     my ($self) = @_;
-    return if testapi::is_serial_terminal();    # in case it is VIRTIO_CONSOLE=1 nothing below make sense
+    return if testapi::is_serial_terminal();    # in case it is NO_VIRTIO_CONSOLE=1 nothing below make sense
```

Comment:
I think it should be `NO_VIRTIO_CONSOLE=0`

Reply:
@cfconrad changed in separate commit (I wanted to be more obvious, that this change depends on https://github.com/os-autoinst/os-autoinst/pull/1045)